addons/hr_contract/models/hr_contract.py

```diff
@@ -147,6 +147,13 @@ def _check_dates(self):
                     contract=contract.name, start=contract.date_start, end=contract.date_end,
                 ))
 
+    @api.onchange('state')
+    def _onchange_state(self):
```

Comment:
Not a fun of doing it in onchange.. I did something similar to it in one of my ongoing tasks..

https://github.com/odoo/odoo/pull/171967/files

What do you think about it? :thinking:

Reply:
Alright, I will try to implement your method and get back to you afterwards.

Reply:
Couple questions though, just to make sure I understand what is going on (I am very new).
- Why the check on contract.job_id != employee.job_id ? Is it just for performance to avoid an unnecessary write ? Why not have that check on department_id ?

Reply:
The change is done and green on runbot